### ISSUE TITLE
BUG with v2converter.ahk latest ver.

### ISSUE THREAD

**Issue by H5820121** [mmikeww/AHK-v2-script-converter#250]
After "Would you like to see the changes made?" -> Yes
![image](https://github.com/user-attachments/assets/14bdb4e5-c708-40e5-9602-00725957e6db)
![image](https://github.com/user-attachments/assets/6cd8479a-e8d1-49bf-802e-396951be1477)

**Comment by Banaanae** [comment]
I can't replicate
Did you use any command line arguments?
If not could you place this line above the fileappend and send the result?
`MsgBox "newPath: [" newPath "]"`

**Comment by andymbody** [comment]
I think that is a line that I added recently (past couple weeks) as part of the fix for some source files having LF instead of CRLF line terminators. The converter creates the temp file whether this is the case or not (just for consistency). That temp file path will be in same folder as the original v1 source file.

Where is this source folder in your case? Is the original v1 source file...
1. on local drive?
2. on a network drive?
3. virtual drive?
4. usb drive? 
5. mapped drive?
6. remote drive?
7. 
As @Banaanae  requested... please provide the path string for newPath.

If v1 script path folder causes issues (is it locked, or not local?) , might need to save the temp file to the local path used for that purpose (common temp folder).

**Comment by Banaanae** [comment]
@andymbody any ideas on why the V2 file worked and the V1 file didn't? Not at home currently but I'm pretty sure the converted V2 is created before and isn't in a try (if it is it should be removed)

**Comment by fade2gray** [comment]
Regarding line endings, why does `ConvertFuncs.ahk` have Unix style line endings?

**Comment by andymbody** [comment]
> any ideas on why the V2 file worked and the V1 file didn't?

Good point... 

The temp file path is derived from the source file path using SplitPath. The converted file path takes into account the CL args and is bound to do a better job. So I should have used FNOut to determine the temp-file path instead.

If we hear back from OP with the path details, we can determine the cause, otherwise we can change the parsing to look at FNOut, and/or use a local temp folder. I think parsing FNOut should work since the v2-converted-file did not generate an exception (I so no TRY, so all good there).

**Comment by andymbody** [comment]
Submitted fix for this #255.

OP, once (if) this fix is merged, please check back to see when (if) this fix is merged into the project. Then download and test please. Thanks!

**Comment by H5820121** [comment]
The version 17 days ago didn't have the bug, the bug was created in some
version later.
The bug is also when everything is on a local disk.
In the conversion itself there is no problem, (by the way, your converter
does the job great! And this is the opportunity to personally thank you and
all the helpers).
The compare in QuickConvertorV2.ahk works correctly, and also in VSC.
The MsgBox says: newPath: [R:\win+E_AHKv1v2_י"אח'195226.ahk]
The name of the original file is R:\win+E.ahk, in newPath strange
characters  י"אח' are added.

‫בתאריך יום א׳, 14 ביולי 2024 ב-7:35 מאת ‪andymbody‬‏ <‪
***@***.***‬‏>:‬

**Comment by andymbody** [comment]
> The MsgBox says: newPath: [R:\win+E_AHKv1v2_י"אח'195226.ahk] The name of the original file is R:\win+E.ahk, in newPath strange characters י"אח' are added

Thanks for this info... this pinpoints the issue.

That trailing number should be the date/time in the following format MMddHHmmss. Clearly this is not being returned correctly. Must be a locale issue. The month should never be 19, the day should never be 52, the hour should never be 26. I'm going to have to investigate why this value is being returned by FormatTime().

**Comment by H5820121** [comment]
@andymbody "Must be a locale issue"
I used when Windows is set to the Hebrew calendar, but now when I set
Windows to the Gregorian calendar there was no bug. Is there a way that
v2converter.ahk will not depend on the type of calendar, and like the
compare in QuickConvertorV2.ahk that works correctly in any calendar type?

‫בתאריך יום א׳, 14 ביולי 2024 ב-21:13 מאת ‪andymbody‬‏ <‪
***@***.***‬‏>:‬

**Comment by andymbody** [comment]
> Is there a way that v2converter.ahk will not depend on the type of calendar

I made a tweak that will discard any invalid filename characters. This should prevent the error going forward. So, no matter which calendar you are using, only digits will be preserved when naming the temp file.

The name used for the temp file is not critical. The date/time is simply used to ensure that the file name is unique enough to avoid over-writing an existing file in the same folder. There are other ways to handle this, so if more issues arise, we can change the way the temp file is handled going forward.

**Comment by H5820121** [comment]
Thank you!
What is the direct link where I can download the new v2converter.ahk file?

‫בתאריך יום א׳, 14 ביולי 2024 ב-22:11 מאת ‪andymbody‬‏ <‪
***@***.***‬‏>:‬

**Comment by andymbody** [comment]
> What is the direct link where I can download the new v2converter.ahk file?

It won't be available on this repo until approved and merged, but if you would rather not wait... download the file here...

https://github.com/andymbody/AHK-v2-script-converter/tree/240714_AMB_Fix_250_3

**Comment by H5820121** [comment]
Now it's working properly, thanks!

‫בתאריך יום א׳, 14 ביולי 2024 ב-22:34 מאת ‪andymbody‬‏ <‪
***@***.***‬‏>:‬

**Comment by Banaanae** [comment]
> There is a better solution, but this was just a quick tweak since my time is limited at the moment to deal with this more effectively.

Should be fine as is, if it does cause issues in the future a static `_newV2CRLF.ahk` would also work

> Regarding line endings, why does `ConvertFuncs.ahk` have Unix style line endings?

It does? My IDE shows CRLF

**Comment by fade2gray** [comment]
@Banaanae

Download the raw `ConvertFuncs.ahk` from the source and check that in your IDE.

If you download the master source zip and did a diff between that and your local clone (I use WinMerge), I think you'll be surprised at the number of files that your IDE reports as having CRLF line endings, are a lot less than you think.

In my case, using, GitKraken, I found the problem was to do with the `.gitconfig` settings file found in the user directory, and because GitKraken couldn't explicitly find the setting ...
```ini
[core]
	autocrlf = false
```
... it automatically assumes `autocrlf = true`, and reconciles all files that have LF endings, to have CRLF endings. I think it may do the same in reverse when pushing files back upstream, but I can't be sure of that. The same may be happening with you using GitUI.

**Comment by Banaanae** [comment]
So it is!
I'll get around to implementing https://github.com/mmikeww/AHK-v2-script-converter/discussions/234#discussioncomment-9977127
In that same PR I'll run a [git renormalize](https://stackoverflow.com/questions/7156694/git-how-to-renormalize-line-endings-in-all-files-in-all-revisions)
See also https://stackoverflow.com/questions/1967370/git-replacing-lf-with-crlf
